odbc/src/api/errors.rs

```diff
@@ -62,6 +62,21 @@ pub enum ODBCError {
         VENDOR_IDENTIFIER
     )]
     FractionalTruncation(String),
+    #[error(
+        "[{}][API] fractional seconds data \"{0}\" was truncated to nanoseconds",
+        VENDOR_IDENTIFIER
+    )]
+    FractionalSecondsTruncation(String),
+    #[error(
+        "[{}][API] seconds data \"{0}\" was truncated to fixed point",
```

Comment:
This error message sounds a little odd to me.
What about "Fractional seconds truncated from \"{0}\"" even maybe simply ""Fractional seconds truncated" so that we are sure not to log any data (I agree in this case we know it is a datetime so it is likely not sensitive)?